cmd/main.go

```diff
@@ -455,6 +473,104 @@ func run(mainConfig Config) (err error) {
+		mgr, err = ctrl.NewManager(&rest.Config{}, mainConfig.options)
+	} else {
+		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), mainConfig.options)
+	}
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		return err
+	}
+	var entryReconciler reconciler.Reconciler
+	if mainConfig.reconcile.ClusterSPIFFEIDs || mainConfig.reconcile.ClusterStaticEntries {
+		entryReconciler = spireentry.Reconciler(spireentry.ReconcilerConfig{
+			TrustDomain:          trustDomain,
+			ClusterName:          mainConfig.ctrlConfig.ClusterName,
+			ClusterDomain:        mainConfig.ctrlConfig.ClusterDomain,
+			K8sClient:            nil,
+			EntryClient:          spireClient,
+			IgnoreNamespaces:     mainConfig.ignoreNamespacesRegex,
+			GCInterval:           mainConfig.ctrlConfig.GCInterval,
+			ClassName:            mainConfig.ctrlConfig.ClassName,
+			WatchClassless:       mainConfig.ctrlConfig.WatchClassless,
+			ParentIDTemplate:     mainConfig.parentIDTemplate,
+			Reconcile:            mainConfig.reconcile,
+			EntryIDPrefix:        mainConfig.ctrlConfig.EntryIDPrefix,
+			EntryIDPrefixCleanup: mainConfig.ctrlConfig.EntryIDPrefixCleanup,
+			StaticManifestPath:   mainConfig.ctrlConfig.StaticManifestPath,
+		})
+	}
+	var federationRelationshipReconciler reconciler.Reconciler
+	if mainConfig.reconcile.ClusterFederatedTrustDomains {
+		federationRelationshipReconciler = spirefederationrelationship.Reconciler(spirefederationrelationship.ReconcilerConfig{
+			K8sClient:          nil,
+			TrustDomainClient:  spireClient,
+			GCInterval:         mainConfig.ctrlConfig.GCInterval,
+			ClassName:          mainConfig.ctrlConfig.ClassName,
+			WatchClassless:     mainConfig.ctrlConfig.WatchClassless,
+			StaticManifestPath: mainConfig.ctrlConfig.StaticManifestPath,
+		})
+		if mainConfig.ctrlConfig.StaticManifestPath == nil {
+			if err = (&controller.ClusterFederatedTrustDomainReconciler{
+				Client:    mgr.GetClient(),
+				Scheme:    mgr.GetScheme(),
+				Triggerer: federationRelationshipReconciler,
+			}).SetupWithManager(mgr); err != nil {
+				setupLog.Error(err, "unable to create controller", "controller", "ClusterFederatedTrustDomain")
+				return err
+			}
+		}
+	}
+	go func() {
+		err = entryReconciler.Run(context.TODO())
+		if err != nil {
+			setupLog.Error(err, "failure starting entry reconciler", "controller", "ClusterStaticEntry")
+		}
+	}()
+	go func() {
+		err = federationRelationshipReconciler.Run(context.TODO())
+		if err != nil {
+			setupLog.Error(err, "failure starting federation relationship reconciler", "controller", "ClusterFederatedTrustDomain")
+		}
+	}()
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		return err
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		return err
+	}
+
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctx); err != nil {
```

Comment:
is it possible to get into an edge case caused by go routines didnt start at before this one? 
if that is the case may we use groups?

Reply:
Yeah. Thats a good thought. Will add

Reply:
Done